File: src/main/java/com/syncstream_dev/syncstream/model/session/PrivateSession.java
package com.syncstream_dev.syncstream.model.session;

import com.syncstream_dev.syncstream.model.ChatStream;
import com.syncstream_dev.syncstream.model.Genre;
import com.syncstream_dev.syncstream.model.VideoInfo;

public class PrivateSession extends Session {
    private char[] password;
    private int maxViewers;

    public PrivateSession(String sessionId, Genre genre, ChatStream chatStream, VideoInfo videoInfo, String password, int maxViewers) {
        super(sessionId, genre, chatStream, videoInfo);
        this.password = password.toCharArray();
        this.maxViewers = maxViewers;
    }

    public char[] getPassword() {
        return password;
    }

    public int getMaxViewers() {
        return maxViewers;
    }
}

File: syncstream/src/datamodules/session/private_session.rs
// private_session.rs implements the PrivateSession module

pub struct PrivateSession {
    info: SessionInfo,
    password: String,
    max_viewers: int,
}

impl PrivateSession {
    pub fn new(
        id: String,
        genre: String,
        participants: Vec<String>,
        chat_stream: String,
        video_info: String,
        password: String,
        max_viewers: int,
    ) -> Self {
        let sessionInfo = SessionInfo::new(id, genre, participants, chat_stream, video_info);
        PrivateSession {
            info: sessionInfo,
            password,
            max_viewers,
        }
    }
}
